packages/preset-uno/src/rules/animation.ts

```diff
@@ -14,4 +15,42 @@ export const animations: Rule[] = [
     return `${keyframes[name]}\n${constructCSS({ animation: `${name} 1s linear infinite` })}`
   }],
   ['animate-none', { animation: 'none' }],
+  [/^animate(?:-duration)?-((\d+)(?:(s|ms)?))$/, ([, d]) => {
+    return {
+      'animation-duration': h.bracket.time(d.replace(/-duration/, '')),
+    }
+  }],
+  [/^animate-delay-((\d+)(?:(s|ms)?))$/, ([, d]) => {
+    return {
```

Comment:
can we support negative values?: for context see https://developer.mozilla.org/en-US/docs/Web/CSS/animation-delay#values

Reply:
should we also force the unit? on the spec it is required and we have `time` with `ms` as default value: it would be better `seconds` since it is an animation?

also, this regex will not support decimals, `2.3s` and `150.25ms` are valid values: https://developer.mozilla.org/en-US/docs/Web/CSS/time

Reply:
time spec from here : https://developer.mozilla.org/en-US/docs/Web/CSS/animation-delay#formal_syntax

Reply:
> can we support negative values?: for context see https://developer.mozilla.org/en-US/docs/Web/CSS/animation-delay#values

I considered this problem when implementing. Multiple groups of attributes only appear when there are multiple groups of animations on the same element. This user scenario is too customized. Even if preset supports multiple groups of attributes, it is still difficult for users to easily complete animation. In this complex scenario, implementing custom `shortcuts` according to requirements should make them easier to implement

Reply:
> should we also force the unit? on the spec it is required and we have `time` with `ms` as default value: it would be better `seconds` since it is an animation?
> 
> also, this regex will not support decimals, `2.3s` and `150.25ms` are valid values: https://developer.mozilla.org/en-US/docs/Web/CSS/time

Excellent point of view! I never used negative numbers, thought there was no scene. Learned from you. it had been solved